### PR TITLE
 REST API tests: patch tasks, post/patch issues

### PR DESCRIPTION
**Pull request by ActiveChooN**
### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2022 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by ActiveChooN** [comment]
@kirill-sizov, can you look at tests, please?

File: tests/rest_api/conftest.py [hunk 1]
```diff
@@ -154,6 +175,22 @@ def check(user_id, jid):
...
+            user_id == assignee_id(jobs[issues[issue_id]['job']])
+    return check
+
+@pytest.fixture(scope='module')
+def is_issue_admin(issues, jobs, is_task_staff):
+    @ownership
+    def check(user_id, issue_id):
+        return is_task_staff(user_id, jobs[issues[issue_id]['job']]['task_id'])
+    return check
+
```
Comment (kirill-sizov) [inline tests/rest_api/conftest.py:193]: Why you decided to separate `issue_admin` and `issue_staff` in such way? Accorging [rego rules](https://github.com/openvinotoolkit/cvat/blob/93ccf2177f560d037fdfad732b98292abfec5944/cvat/apps/iam/rules/issues.rego#L97-L115): if some user is a `issue_admin` then this user automaticly is a `issue_staff`

**Review by kirill-sizov — COMMENTED** [review]
Tests mostly looks nice to me. I have only one question, see bellow.

File: tests/rest_api/test_issues.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        ('supervisor', False, False), ('worker', False, False),
+        ('maintainer', True, True), ('owner',  True, True),
+        ('supervisor', True, True), ('worker', True, True)
+    ])
+    def test_member_create_issue(self, org, role, job_staff, is_allow,
+            find_job_staff_user, find_users, jobs_by_org):
+        users = find_users(role=role, org=org)
+        jobs = jobs_by_org[org]
+        username, jid = find_job_staff_user(jobs, users, job_staff)
+        job, = filter(lambda job: job['id'] == jid, jobs)
```
Comment (kirill-sizov) [inline tests/rest_api/test_issues.py:66]:
```suggestion
    def test_member_create_issue(self, org, role, job_staff, is_allow,
            find_job_staff_user, find_users, jobs_by_org, jobs):
        users = find_users(role=role, org=org)
        username, jid = find_job_staff_user(jobs_by_org[org], users, job_staff)
        job = jobs[jid]
```

A little suggestion for avoiding filtering where it's possible.

**Review by nmanovic — APPROVED** [review]

**Pull request merged**